bash3/read_file.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+
+# Read a file from file path (given as a command line arg),
+# print line by line with line numbers
+
+file_path="$1"
+
+if [ ! -f "$file_path" ]; then
+  echo "File not found: $file_path"
+  exit 1
+fi
+
+i=1
+while IFS= read -r line; do
+  uppercase_line=$(echo "$line" | tr '[:lower:]' '[:upper:]')
+  echo "$i $uppercase_line"
+  ((i++))
+done < "$file_path"
```

Comment:
I think I missed this in my first review. We should be able to do the same thing we do in `stdin.sh` except by reading from the file instead of from stdin. Like this:

```bash
i=1

while IFS= read -r line; do
  echo "$((i++)) $(tr '[:lower:]' '[:upper:]' <<< "$line")"
done < "$file_path"
```